l10n_br_fiscal/models/document.py

```diff
@@ -211,36 +212,103 @@ def _compute_amount(self):
 
     partner_legal_name = fields.Char(
         string='Legal Name',
+        related='partner_id.legal_name',
     )
 
     partner_name = fields.Char(
         string='Name',
+        related='partner_id.name',
     )
 
     partner_cnpj_cpf = fields.Char(
         string='CNPJ',
+        related='partner_id.cnpj_cpf',
     )
 
     partner_inscr_est = fields.Char(
         string='State Tax Number',
+        related='partner_id.inscr_est',
+    )
+
+    partner_ind_ie_dest = fields.Selection(
+        selection=NFE_IND_IE_DEST,
+        string='Contribuinte do ICMS',
+        related='partner_id.ind_ie_dest',
     )
 
     partner_inscr_mun = fields.Char(
         string='Municipal Tax Number',
+        related='partner_id.inscr_mun',
     )
 
     partner_suframa = fields.Char(
         string='Suframa',
+        related='partner_id.suframa',
     )
 
     partner_cnae_main_id = fields.Many2one(
         comodel_name='l10n_br_fiscal.cnae',
         string='Main CNAE',
+        related='partner_id.cnae_main_id',
     )
 
     partner_tax_framework = fields.Selection(
         selection=TAX_FRAMEWORK,
         string='Tax Framework',
+        related='partner_id.tax_framework',
+    )
+
+    partner_street = fields.Char(
+        string='Partner Street',
+        related='partner_id.street',
+    )
+
+    partner_number = fields.Char(
+        string='Partner Number',
+        related='partner_id.street_number',
+    )
+
+    partner_street2 = fields.Char(
+        string='Partner Street2',
+        related='partner_id.street2',
+    )
+
+    partner_district = fields.Char(
+        string='Partner District',
+        related='partner_id.district',
+    )
+
+    partner_country_id = fields.Many2one(
+        comodel_name='res.country',
+        string='Partner Country',
+        related='partner_id.country_id',
+    )
+
+    partner_state_id = fields.Many2one(
+        comodel_name='res.country.state',
+        string='Partner State',
+        related='partner_id.state_id',
+    )
+
+    partner_city_id = fields.Many2one(
+        comodel_name='res.city',
+        string='Partner City',
+        related='partner_id.city_id',
+    )
+
+    partner_zip = fields.Char(
+        string='Partner Zip',
+        related='partner_id.zip',
+    )
+
+    partner_phone = fields.Char(
+        string='Partner Phone',
+        related='partner_id.phone',
+    )
+
+    partner_is_company = fields.Boolean(
+        string='Partner Is Company?',
+        related='partner_id.is_company',
```

Comment:
Esses campos related podem trazer problemas quando documentos fiscais forem importados de sistemas externos.

O documento fiscal deve ser visualizado no sistema no modelo de documento fiscal antes da sua devida escrituração, quando aprovado pelo responsável fiscal então caso necessário a criação ou sincronização dos cadastros deve ser feita.

Reply:
@mileo com todo respeito, **eu discordo categoricamente** e ja conversamos sobre no Telegram...

Pois no framework de importacao que eu bolei pelo generateDS desde o PR #753 de dezembro, **NAO TEM NEHNUM WRITE** de registro existente. E **o unico create** eh aqui:
https://github.com/akretion/l10n-brazil/blob/12.0-l10n_br_nfe-cherrypick-hacks/spec_driven_model/models/spec_import.py#L253
Eh vai que quando se encontra o record_id nesse metodo `match_or_create_m2o` poderia eventualmente ter algum write para atualizar alguns dados mas ate agora nem foi preciso.

Caso tivesse e naquele create super localizados, eh muito facil botar algum filtro configuravel pelo contexto para permitir criar ou atualizar alguns objetos specificos de acordo com o contexto exacto de uso (que pode nao ser apenas importacao de NFe de fornecedor, mas pode tb ser migracao de sistema, dados de demo etc...)

Segundo, como ja falei, eh trivial em 5 linhas de fazer um contexto **DRY RUN** no qual nem esse create seria chamado e ao invez disso, toda logica que acontecer de forma recursiva e automatica poderia ser logada num dicionario/json para retornar warnings e sugerir validações pro usuario/responsável fiscal/qualquer term pomposo. Apos eventuais correcoes do dicionario de importacao da NFe (feita atraves dos wizards que vc bem entender nos modulos que bem entender de acordo com o contexto de uso), ai sim ocoreria a importacao de vdd. Com o framework que eu bolei, eh trivial vc montar um wizard ou objeto qualquer para vizualizar algum documento electronico, nao precisa zoar os modelos fiscais centrais para esse caso de uso.

Entao, como vc ve eh trivial lidar com as objeções que vc faz.

Agora pelo contrario, vc duplicar 20 campos de forma arbitraria  - e porque tb nao duplicar tudo que for dos produtos ou da transportadora? -> eh arbitrario sim! - isso eh uma aumentação da entropia do código, eh igual vc fazer a pasta de dente sair do tubo e esperar que por algum milagro ela vai entrar no tubo de volta de forma espontânea.

Accredite nossa longa experiencia com os mais variados conectores Odoo, **VC TEM QUE TER UMA PUTA DE UMA BOA RAZAO VALIDA PARA DUPLICAR UM MODELO DE DADOS**.
Pois depois, vc parte para discrepância de dados entre o que o usuário vai corrigir na tela do documento fiscale deixar de atualizar nos proprios registros originais, vc vai obrigar a engessar uma porrada de codigo de mapping e de sincronizacao de dados entre esses campos e os campos originais, gerenciar as discrepâncias.

Esse tipo de codigo Frankenstein ja começou a ser feito por vcs sem pena. Partiram do meu PR #753 que tava limpinho com a excecao assumida da visao automatica e comecaram a enfiar esse tipo de codigo de mapping sem pena:
https://github.com/akretion/l10n-brazil/blob/12.0-l10n_br_nfe-cherrypick-hacks/l10n_br_nfe/models/document.py#L503

Isso ja foi um pesadelo de acertar na producao, e ja quebrou 70% das coisas que eu tinha feito na importacao no PR #753 em especial pelo motivos de ter os campos duplicado.